sbtStudentCommands/Pssr.scala.template

```diff
@@ -36,18 +37,20 @@ object Pssr {
     savedExercises
   }
 
-  def pullSolution = Command.command("pullSolution") { state =>
+  def pullSolution: Command = Command.command("pullSolution") { state =>
     val currentExercise = getCurrentExercise(state)
-    val currentExerciseMain = new sbt.File(getRootFolder(state), "exercises/src/main")
-    val cueFolder = getCueFolder(state)
-    val solFolder = new sbt.File(cueFolder, currentExercise + "/src/main")
-    IO.delete(currentExerciseMain)
-    IO.copyDirectory(solFolder, currentExerciseMain)
-    Console.println(Console.GREEN + "[INFO] " + Console.RESET + s"Solution for exercise $currentExercise pulled successfully")
-    state
+    withZipFile(state, currentExercise) { () =>
+      val currentExerciseMain = new sbt.File(getRootFolder(state), "exercises/src/main")
+      val cueFolder = getCueFolder(state)
+      val solFolder = new sbt.File(cueFolder, currentExercise + "/src/main")
+      IO.delete(currentExerciseMain)
+      IO.copyDirectory(solFolder, currentExerciseMain)
+      Console.println(Console.GREEN + "[INFO] " + Console.RESET + s"Solution for exercise $currentExercise pulled successfully")
```

Comment:
Really like the idea!

Just verified the 'principle':

```
  def greenString(msg: String): String = {
    s"${Console.GREEN}$msg${Console.RESET}"
  }

  def pullSolution: Command = Command.command("pullSolution") { state =>
    val currentExercise = getCurrentExercise(state)
    withZipFile(state, currentExercise) { () =>
      val currentExerciseMain = new sbt.File(getRootFolder(state), "exercises/src/main")
      val cueFolder = getCueFolder(state)
      val solFolder = new sbt.File(cueFolder, currentExercise + "/src/main")
      IO.delete(currentExerciseMain)
      IO.copyDirectory(solFolder, currentExerciseMain)
      Console.println(s"${greenString("[INFO]")} Solution for exercise $currentExercise pulled successfully !")
      state
    }
  }
```
I would suggest that we create a new issue for this and let this PR go through as-is.
This way we can continue with processing the different todo's for all courses on Trello cards.